usaspending_api/download/helpers/download_annotation_functions.py

```diff
@@ -620,3 +621,48 @@ def subaward_annotations():
         "prime_award_latest_action_date_fiscal_year": FiscalYear("award__latest_transaction__action_date"),
     }
     return annotation_fields
+
+
+def disaster_recipient_annotations():
+    annotation_fields = {
+        "award_obligations": Coalesce(
+            Sum(
+                "award__financial_set__transaction_obligated_amount",
+                filter=Q(
+                    filter_by_latest_closed_periods("award__financial_set__"),
+                    award__financial_set__disaster_emergency_fund__group_name="covid_19",
+                    award__financial_set__submission__reporting_period_start__gte=str(datetime.date(2020, 4, 1)),
+                ),
+                output_field=DecimalField(),
+            ),
+            0,
+        ),
+        "award_outlays": Coalesce(
+            Sum(
+                "award__financial_set__gross_outlay_amount_by_award_cpe",
+                filter=Q(
+                    filter_by_latest_closed_periods("award__financial_set__"),
+                    award__financial_set__disaster_emergency_fund__group_name="covid_19",
+                    award__financial_set__submission__reporting_period_start__gte=str(datetime.date(2020, 4, 1)),
+                ),
+                output_field=DecimalField(),
+            ),
+            0,
+        ),
+        # Hacky way to get the Sum "total_loan_value" from each unique Award
+        "face_value_of_loans": Coalesce(
+            Sum(
+                Subquery(
+                    Award.objects.filter(financial_set__award_id=OuterRef("award_id"))
+                    .annotate(loan_value_fraction=F("total_loan_value") / Count("id", output_field=DecimalField()))
```

Comment:
![](https://media.giphy.com/media/UnTC9o2HMyUta/giphy.gif)

Reply:
@kbard did you think of this as viable instead of the 13 CTEs?

Reply:
Does this actually work?  I tried something similar when I was working on agency loans but the math didn't work out because there were multiple awards per agency and they were repeated.  I'd have to materialize this to SQL to see what's it actually doing.

Reply:
```
...
    COALESCE(SUM((SELECT (U0."total_loan_value" / COUNT(U0."id")) AS "loan_value_fraction"
FROM "awards" U0
INNER JOIN "financial_accounts_by_awards" U1 ON (U0."id" = U1."award_id")
...
```